sets/seaside.py

```diff
@@ -60,10 +60,72 @@ def duration(self):
 		self.game.announce("-- drawing {}".format(drawn))
 
 
+class Treasure_Map(crd.Card):
+	def __init__(self, game, played_by):
+		crd.Card.__init__(self, game, played_by)
+		self.title = "Treasure Map"
+		self.price = 4
+		self.description = "Trash this and another copy of Treasure Map from your hand." \
+		                   " If you do trash two Treasure Maps, gain 4 Gold cards, putting them on top of your deck."
+		self.type = "Action"
+
+	@gen.coroutine
+	def play(self, skip=False):
+		crd.Card.play(self, skip)
+		if self.played_by.hand.get_count('Treasure Map') > 0:
+			selection = yield self.played_by.select(1, 1, ["Yes", "No"], "Would you like to trash "
+			            "this and another copy of treasure map from hand to gain 4 Gold to the top of your deck?")
+			if selection[0] == 'Yes':
+				self.game.trash_pile.append(self.played_by.played_cards.pop())
+				self.played_by.discard(['Treasure Map'], self.game.trash_pile)
+				self.game.update_trash_pile()
```

Comment:
should yield the self.played_by.discard since its a coroutine even though it shouldn't trigger